tests/types/test_numpy.py

```diff
@@ -0,0 +1,379 @@
+def test_dump_numpy_ulonglong(numpy_conn, val, expr, fmt_in):
+
+    val = np.ulonglong(val)
+
+    assert isinstance(val, np.ulonglong)
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
+
+    cur.execute(f"select {expr} = %{fmt_in.value}", (val,))
+    assert cur.fetchone()[0] is True
+
+
+# Test float special values
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (np.PZERO, "'0.0'::float8"),
+        (np.NZERO, "'-0.0'::float8"),
+        (np.nan, "'NaN'::float8"),
+        (np.inf, "'Infinity'::float8"),
+        (np.NINF, "'-Infinity'::float8"),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_special_values(numpy_conn, val, expr, fmt_in):
+    # All numpy special values are python float types,
+    # So the Dumper used should be the normal float dumper
+
+    if val == np.nan:
+        assert np.nan == np.NAN == np.NaN
+
+    if val == np.inf:
+        assert np.inf == np.Inf == np.PINF == np.infty
+
+    assert isinstance(val, float)
+
+    cur = numpy_conn.cursor()
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+
+    assert cur.fetchone()[0] is True
+
+    cur.execute(f"select {expr} = %{fmt_in.value}", (val,))
+    assert cur.fetchone()[0] is True
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (31e3, "31e3::float4"),
+        (31e-3, "31e-3::float4"),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_float16(numpy_conn, val, expr, fmt_in):
+
+    val = np.float16(val)
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
+
+    cur.execute(
+        f"select abs(({expr}::float4 - %{fmt_in.value}) / {expr}::float4)", (val,)
+    )
+    res = cur.fetchone()[0]
+
+    assert res <= 1e-3
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (256e6, "256e6::float4"),
+        (256e-6, "256e-6::float4"),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_float32(numpy_conn, val, expr, fmt_in):
+
+    val = np.float32(val)
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
+
+    cur.execute(
+        f"select abs(({expr}::float4 - %{fmt_in.value}) / {expr}::float4)", (val,)
+    )
+    res = cur.fetchone()[0]
+
+    assert res <= 1e-6
+
+
+@pytest.mark.parametrize(
+    "val, expr",
+    [
+        (2048e32, "2048e32::float8"),
+        (2048e-32, "2048e-32::float8"),
+    ],
+)
+@pytest.mark.parametrize("fmt_in", PyFormat)
+def test_dump_numpy_float64(numpy_conn, val, expr, fmt_in):
+
+    val = np.float64(val)
+
+    cur = numpy_conn.cursor()
+
+    cur.execute(f"select pg_typeof({expr}) = pg_typeof(%{fmt_in.value})", (val,))
+    cur.fetchone()[0] is True
```

Comment:
Missing assert here and in other places
```suggestion
    assert cur.fetchone()[0] is True
```